src/components/StatusButton/__test__/index.test.jsx

```diff
@@ -0,0 +1,42 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import StatusButton from "..";
+
+const MockStatusButton = () => {
+  return <StatusButton id="3456" />;
+};
+
+describe("Status button component", () => {
+  it("should render status button correctly", () => {
+    render(<StatusButton />);
+    const statusButton = screen.getByTestId("active-status");
+    expect(statusButton).toBeInTheDocument();
+  });
+
+  it("on hover show status list", () => {
+    render(<StatusButton />);
+    const dropdown = screen.getByTestId("dropdown");
+    const dropdownContent = screen.getByTestId("dropdown-content");
+    fireEvent.mouseOver(dropdown);
+    expect(dropdownContent).toBeVisible();
+  });
+
+  it("should have four buttons", () => {
+    render(<StatusButton />);
+    const buttonList = screen.getAllByRole("button");
+    expect(buttonList).toHaveLength(4);
+  });
+
+  it("should not have check icon", () => {
+    render(<StatusButton />);
+    const checkIcon = screen.queryByTestId("check-icon");
+    expect(checkIcon).not.toBeInTheDocument();
+  });
+
+  it("on click should show check icon", () => {
```

Comment:
you can add one more check if value/id is passed as a prop then that value is selected